cmd/kube-apiserver/app/options/options.go

```diff
@@ -50,18 +50,21 @@ type ServerRunOptions struct {
 	APIEnablement           *genericoptions.APIEnablementOptions
 	EgressSelector          *genericoptions.EgressSelectorOptions
 
-	AllowPrivileged           bool
-	EnableLogsHandler         bool
-	EventTTL                  time.Duration
-	KubeletConfig             kubeletclient.KubeletClientConfig
-	KubernetesServiceNodePort int
-	MaxConnectionBytesPerSec  int64
+	AllowPrivileged            bool
+	EnableLogsHandler          bool
+	EventTTL                   time.Duration
+	KubeletConfig              kubeletclient.KubeletClientConfig
+	KubernetesServiceNodePort  int
+	KubernetesServiceClusterIP string
+	MaxConnectionBytesPerSec   int64
 	// ServiceClusterIPRange is mapped to input provided by user
 	ServiceClusterIPRanges string
 	//PrimaryServiceClusterIPRange and SecondaryServiceClusterIPRange are the results
 	// of parsing ServiceClusterIPRange into actual values
 	PrimaryServiceClusterIPRange   net.IPNet
 	SecondaryServiceClusterIPRange net.IPNet
+	// APIServerServiceIP is the result of parsing KubernetesServiceClusterIP
```

Comment:
This pair of names is confusing. `KubernetesServiceClusterIPString` and `KubernetesServiceClusterIP` are better, or use `...IP` and `...ParsedIP`, something like that. But don't call the same thing both a service IP and a cluster IP.